credsweeper/utils/util.py

```diff
@@ -73,8 +73,23 @@ def get_shannon_entropy(cls, data: str, iterator: Chars) -> float:
 
     @classmethod
     def read_patch_file(cls, path: str) -> List[str]:
-        with open(path, "r") as patch_file:
-            diff_data = patch_file.readlines()
+        try:
+            with open(path, "r", encoding="utf8") as patch_file:
+                diff_data = patch_file.readlines()
+        except UnicodeDecodeError:
+            logging.warning(f"UnicodeDecodeError: Can't read patch content from \"{path}\" as UTF-8.")
+            try:
+                logging.info(f"Try to open patch file:\"{path}\" as UTF-16")
+                with open(path, "r", encoding="utf16") as patch_file:
+                    diff_data = patch_file.readlines()
+            except UnicodeError:
+                logging.warning(f"UnicodeError: Can't read patch content from \"{path}\" as UTF-16.")
+                try:
+                    logging.info(f"Try to open patch file:\"{path}\" as Latin-1")
+                    with open(path, "r", encoding="latin_1") as patch_file:
+                        diff_data = patch_file.readlines()
+                except Exception as e:
+                    logging.error(f"Unexpected Error: Can't read patch content from \"{path}\". Error message: {e}")
```

Comment:
I think it might be better to do something like

```python3
supported_encoding = ["utf8", ... ]
for encoding in supported_encoding:
    try:
        ...
    except UnicodeDecodeError:
        ...
    except UnicodeError:
        ...
```

Current exception-in-exception architecture is a bit hard to read